lib/archethic/metrics/helpers.ex

```diff
@@ -0,0 +1,381 @@
+defmodule ArchEthic.Metrics.Helpers do
+  @moduledoc """
+  Provides helper methods & data in transformation of metrics.
+  """
+  alias ArchEthic.DB
+  require Logger
+
+  def network_collector() do
+    data = Enum.filter(retrieve_network_metrics(), fn {key, _val} -> req_metrics(key) end)
+    data = Enum.into(data, %{})
+    Map.put(data, "tps", DB.get_latest_tps())
```

Comment:
I think the TPS should come from the metrics and not only from the DB.
This one is computed for an entire day or self-repair interval, which may not reflects the real load

Reply:
**Calculating The Transaction per second **
archethic_mining_full_transaction_validation_duration_bucket{le="0.01"} 0
archethic_mining_full_transaction_validation_duration_bucket{le="0.025"} 0
archethic_mining_full_transaction_validation_duration_bucket{le="0.05"} 470
archethic_mining_full_transaction_validation_duration_bucket{le="0.1"} 1219
archethic_mining_full_transaction_validation_duration_bucket{le="0.5"} 1471
archethic_mining_full_transaction_validation_duration_bucket{le="0.8"} 1510
archethic_mining_full_transaction_validation_duration_bucket{le="1"} 1511
archethic_mining_full_transaction_validation_duration_bucket{le="1.2"} 1512
archethic_mining_full_transaction_validation_duration_bucket{le="1.5"} 1512
archethic_mining_full_transaction_validation_duration_bucket{le="2"} 1512
archethic_mining_full_transaction_validation_duration_bucket{le="2.5"} 1512
archethic_mining_full_transaction_validation_duration_bucket{le="3"} 1512
archethic_mining_full_transaction_validation_duration_bucket{le="5"} 1512
archethic_mining_full_transaction_validation_duration_bucket{le="10"} 1512
archethic_mining_full_transaction_validation_duration_bucket{le="+Inf"} 1512
archethic_mining_full_transaction_validation_duration_sum 162.63586202400006
archethic_mining_full_transaction_validation_duration_count 1512


So to calculate Transaction per second :   
archethic_mining_full_transaction_validation_duration_sum 162.63586202400006
archethic_mining_full_transaction_validation_duration_count 1512

total no of events in less than 10 ms is 1512
So total number of events =1512 
and total time 162 ms
.162 seconds  --> 1512 txns
in 1s     --> 1512/ .162
==> 9333 txn/s
is this maths correct ?

Reply:
hi Sir , please mention how to calculate  Transaction Per second ? from /metrics endpoint .  .???
@samuel-uniris

Reply:
1512 is the number of transaction validation which take less than 10s, from the last polling interval (i.e 5 seconds)
162 is total of the observed values (sum of the milliseconds)
So the TPS would be: 1512/162: 9.33

Reply:
Hi @samuel-uniris  Sir,
I have to add all the  txn per second of all the nodes. (and not do 1 by average total validation time of network.)

Reply:
I think you can compute the TPS for the all the nodes and then do the average